tools/integration_tests/util/operations/validation_helper.go

```diff
@@ -29,8 +29,8 @@ func ValidateNoFileOrDirError(path string, t *testing.T) {
 }
 
 func CheckErrorForReadOnlyFileSystem(err error, t *testing.T) {
-	if strings.Contains(err.Error(), "read-only file system") || strings.Contains(err.Error(), "permission denied") || strings.Contains(err.Error(), "Permission denied") {
+	if err != nil && (strings.Contains(err.Error(), "read-only file system") || strings.Contains(err.Error(), "permission denied") || strings.Contains(err.Error(), "Permission denied")) {
```

Comment:
Nit: we can check the `nil` on the top and return with proper error logging.

Reply:
Done